pkg/cache/v3/simple.go

```diff
@@ -286,7 +286,7 @@ func superset(names map[string]bool, resources map[string]types.ResourceWithTTL)
 }
 
 // CreateWatch returns a watch for an xDS request.
-func (cache *snapshotCache) CreateWatch(request *Request, value chan Response) func() {
+func (cache *snapshotCache) CreateWatch(request *Request, streamState *stream.StreamState, value chan Response) func() {
```

Comment:
IMO I think it would be better if we didn't make this a pointer. I see your point but I'd like to match the external cache API to Delta so users don't get confused. `CreateDeltaWatch` doesn't take a pointer so it would make sense for this to not either.